holo-ospf/src/ospfv2/lsdb.rs

```diff
@@ -346,6 +353,44 @@ impl LsdbVersion<Self> for Ospfv2 {
                 }
             }
         }
+
+        // examine for DynamicHostnameTlv
```

Comment:
you could try this to avoid the code duplication (area/AS scope):
```rs
        // Update hostname database.
        if let LsaBody::OpaqueArea(LsaOpaque::RouterInfo(router_info))
        | LsaBody::OpaqueAs(LsaOpaque::RouterInfo(router_info)) = &lsa.body
        {
            if let Some(hostname_tlv) = router_info.info_hostname.as_ref() {
                // Install or update hostname.
                instance
                    .state
                    .hostnames
                    .insert(lsa.hdr.adv_rtr, hostname_tlv.hostname.clone());
            } else {
                // Remove hostname if it exists.
                instance.state.hostnames.remove(&lsa.hdr.adv_rtr);
            }
        }
```